modules/core/shared/src/main/scala/gem/Step.scala

```diff
@@ -16,32 +19,140 @@ object Step {
 
   // TODO: better name for this
   sealed trait Base
-  object Base {
+
+  object Base extends BaseOptics {
+
     final case object Bias extends Base
     final case object Dark extends Base
     final case class  Gcal(gcal: GcalConfig) extends Base
     final case class  Science(telescope: TelescopeConfig) extends Base
     final case class  SmartGcal(smartGcalType: SmartGcalType) extends Base
+
+    implicit val EqBias: Eq[Bias.type] =
+      Eq.allEqual
+
+    implicit val EqDark: Eq[Dark.type] =
+      Eq.allEqual
+
+    object Gcal extends GcalOptics {
+
+      implicit val EqGcal: Eq[Gcal] =
+        Eq.by(_.gcal)
+
+    }
+
+    trait GcalOptics {
+
+      /** @group Optics */
+      val gcal: Lens[Gcal, GcalConfig] =
+        Lens[Gcal, GcalConfig](_.gcal)(a => _.copy(gcal = a))
+
+    }
+
+    object Science extends ScienceOptics {
+
+      implicit val EqScience: Eq[Science] =
+        Eq.by(_.telescope)
+
+    }
+
+    trait ScienceOptics {
+
+      /** @group Optics */
+      val telescope: Lens[Science, TelescopeConfig] =
+        Lens[Science, TelescopeConfig](_.telescope)(a => _.copy(telescope = a))
+
+    }
+
+    object SmartGcal extends SmartGcalOptics {
+
+      implicit val EqSmartGcal: Eq[SmartGcal] =
+        Eq.by(_.smartGcalType)
+
+    }
+
+    trait SmartGcalOptics {
+
+      /** @group Optics */
+      val smartGcalType: Lens[SmartGcal, SmartGcalType] =
+        Lens[SmartGcal, SmartGcalType](_.smartGcalType)(a => _.copy(smartGcalType = a))
+
+    }
+
+    implicit val EqBase: Eq[Base] =
+      Eq.instance {
+        case (a: Bias.type, b: Bias.type) => a === b
+        case (a: Dark.type, b: Dark.type) => a === b
+        case (a: Gcal,      b: Gcal     ) => a === b
+        case (a: Science,   b: Science  ) => a === b
+        case (a: SmartGcal, b: SmartGcal) => a === b
+        case _                            => false
+      }
```

Comment:
Is this the right way to do `Eq` for an ADT?  I did the same thing for `DynamicConfig`.

Reply:
At least I do the same for Eq in an ADT

Reply:
Me too, except that I make the `object` cases return always `true`.